### ISSUE TITLE
Fail under Windows 10 x64

### ISSUE THREAD

**Issue by wtask** [gofrs/uuid#69]
After converting my own project into go.mod and run `go test all` under module root, got:
```
--- FAIL: TestGenerator (0.02s)
--- FAIL: TestGenerator/NewV2 (0.00s)
--- FAIL: TestGenerator/NewV2/DifferentAcrossCalls (0.00s)
generator_test.go:210: generated identical UUIDs across calls: 3912a521-2f04-21e9-8002-309c238cfc62
FAIL
FAIL github.com/gofrs/uuid 0.757s
```

**Comment by CameronAckermanSEL** [comment]
Thanks so much for reporting! I've confirmed the issue, and it looks like the root cause is on windows, the functions that return the uid and gid return -1

in `generator.go`

```var (
	posixUID = uint32(os.Getuid())
	posixGID = uint32(os.Getgid())
)```

Then in `os`

```
// Getuid returns the numeric user id of the caller.
//
// On Windows, it returns -1.
func Getuid() int { return syscall.Getuid() }

// Geteuid returns the numeric effective user id of the caller.
//
// On Windows, it returns -1.
func Geteuid() int { return syscall.Geteuid() }

// Getgid returns the numeric group id of the caller.
//
// On Windows, it returns -1.
func Getgid() int { return syscall.Getgid() }
```

**Comment by zerkms** [comment]
Isn't uid/gid just first 4 bytes and the rest still should be random?

**Comment by CameronAckermanSEL** [comment]
For posterity: 
http://pubs.opengroup.org/onlinepubs/9668899/chap5.htm#tagcjh_08_02_01_01

https://tools.ietf.org/html/rfc4122#section-4.1

**Comment by theckman** [comment]
@wtask we've discussed this a bit and we believe this to be an artifact of the V2 UUID itself, and not a bug or issue with the implementation. The quote in https://github.com/gofrs/uuid/issues/69#issuecomment-462948618 from @zerkms comes from:

- https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_2_(date-time_and_MAC_address,_DCE_security_version)

The immediate "fix" is to likely add a sleep statement to the test to give enough time for the internal timestamp value to change. Secondarily, to that we should add some documentation around the limitations of the V2 UUID so someone is not surprised.

Long-term we may remove the V2 UUID since it does seem to have some issues, and is not part of RFC-4122.

**Comment by wtask** [comment]
Thanks for replies, guys! 
First, I'm propably will unable to check fixes due to `uuid` is an indirect dependency for my module:

```
github.com/gofrs/uuid v3.2.0+incompatible // indirect
```

Also, I think you should not release a fix by adding crazy sleep over 7 seconds into the test. It is nothing fixes under Windows, only hides a problem with package for this OS. 
In my opinion, if the only problem is that UID / GID are not available in WIndows, then you need to calculate conditional values for them, for example, in the init() function instead of directly referring to the os package. For example, if it is allowed, calculate crc32 for username and hostname on Windows. And so on...

```
C:\dev\go>whoami
msi\mike

C:\dev\go>echo %username%
Mike

C:\dev\go>hostname
MSI
```

**Comment by zerkms** [comment]
@wtask it does not add a sleep to test, it just shifts time in a time provider.

> It is nothing fixes under Windows, only hides a problem with package for this OS

As we discovered yesterday, it works as designed, UUID v2 is not capable of generating unique UUIDs more frequently than once every 7 seconds (it's taken as granted from wikipedia, needs a "scientific" confirmation).

**Comment by wtask** [comment]
... also you may have an overflow for uint32(-1) and you should simply use predefined positive values, not -1 in both cases

**Comment by zerkms** [comment]
overflow for a what value?

**Comment by wtask** [comment]
-1 overflows uint32 (undefined)

**Comment by theckman** [comment]
@wtask I'm not sure I see it as a problem, but that could just be me missing something. With `-1` overflowing `uint32`, it effectively becomes a positive predefined value: `4294967295`.

**Comment by zerkms** [comment]
In Go there are no undefined integral overflows.

**Comment by wtask** [comment]
The extra bits, including the sign bit, are simply discarded, and the result in runtime is not predefined, but depends on the compiler as I know. But literal expression `i := uint32(-1)` simply does not compile. But if any indefinite positive integer is valid `uuid` package - ok.

**Comment by theckman** [comment]
@wtask I've figured out why this is failing, and it's pretty great. Your system is running the tests _too fast_.

In the test that failed we pass in `uuid.DomainOrg` as the `domain` to the `NewV2` function. When you use `DomainOrg` the UUID generation code _does not_ overwrite the timestamp field on the UUID. The Timestamp itself is a 60-bit unsigned integer, that is the number of 100-nanosecond increments since some epoch value in the 1500s.

So if you are able to generate two UUIDs within the same 100ns bucket of time, this test will fail. If your system is slower, like I suspect TravisCI to be, the tests will pass just fine. So I think the fix would be to underclock your workstation. :wink:

All jokes aside, it's an interesting side effect. I think pursuing a fix similar to what we did in #70, but with a tighter timeframe, makes sense.
